ldapauthenticator/ldapauthenticator.py

```diff
@@ -97,17 +97,26 @@ def authenticate(self, handler, data):
         if password is None or password.strip() == '':
             self.log.warn('Empty password')
             return None
-
-        userdn = self.bind_dn_template.format(username=username)
-
-        server = ldap3.Server(
-            self.server_address,
-            port=self.server_port,
-            use_ssl=self.use_ssl
-        )
-        conn = ldap3.Connection(server, user=userdn, password=password)
-
-        if conn.bind():
+        
+        isBound = False
+        for dn in self.bind_dn_template:
+            #self.log.debug("LOOPING DN")
+            userdn = dn.format(username=username)
+            self.log.debug("DN: '%s'", userdn)
+            server = ldap3.Server(
+                self.server_address,
+                port=self.server_port,
+                use_ssl=self.use_ssl
+            )
+            self.log.debug("GET LDAP CONNECTION FOR USER: '%s'", username)
+            conn = ldap3.Connection(server, user=userdn, password=password)
+            self.log.debug("GOT LDAP CONNECTION FOR USER: '%s'", conn)
+            isBound = conn.bind()
+            self.log.debug("CONN_BIND: "+ str(isBound) + ":" + username ) 
```

Comment:
For all logging, you can use format strings instead of explicit concatenation. 

```python
self.log.debug('Attempting to bind with dn %s', dn)
```